### DISCUSSION TITLE
ESM vs Common JS modules

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#260] [Development]
Chromium 83 already fully supports ESM modules. Node is getting there with v14 (although we are not getting to node 14)

But if we use bundler/transpile why not to use latest language features?

**Comment by tobias-klein** [comment]
I agree that we can do this as long as the bundler works. I guess this could be an argument against Browserify, but I didn't check yet whether this has changed in the most recent versions. We would need to further analyze this. It's quite some refactoring effort considering the large amount of files we're already dealing with.

**Reply by zhuiks** [reply]
If we do refactor with web components, new components can be ESM modules. That is the point of this proposal

**Reply by tobias-klein** [reply]
I think this is unrelated/independent to the web components discussion. A web component can also be served as a regular Node.js module.
